comet/broker.py

```diff
@@ -644,25 +649,28 @@ async def get_dataset(ds_id, wait=True):
     Returns
     -------
     Dataset
-        The dataset from cache or redis. `None` if the dataset doesn't exist and wait was `False`.
+        The dataset from cache or redis.
 
     Raises
     ------
-    TimeoutError
-        If waiting for the dataset timed out.
+    DatasetNotFoundError
+        If the dataset doesn't exist or waiting for the dataset timed out.
+
     """
     if wait:
         # Check if existing and wait if not
         found = await wait_for_dset(ds_id)
         if not found:
-            raise TimeoutError("Dataset {} not found: Timeout.".format(ds_id))
+            raise DatasetNotFoundError("Dataset {} not found: Timeout.".format(ds_id))
```

Comment:
Is there any reason you would still need to differentiate between a time out and the dataset actually not being found?